Dockerfile

```diff
@@ -1,5 +1,5 @@
-FROM openjdk
+FROM openjdk:8-jdk-alpine
 WORKDIR /
-ADD target/conter-0.0.1-SNAPSHOT.jar conter-0.0.1-SNAPSHOT.jar
+COPY target/conter-0.0.1-SNAPSHOT.jar app.jar
 EXPOSE 8080
-CMD java -jar conter-0.0.1-SNAPSHOT.jar
+ENTRYPOINT [ "sh", "-c", "java -Dutopia.search.API=$1 -Dutopia.booking.API=$2 -Dutopia.cancellation.API=$3 -Djava.security.egd=file:/dev/./urandom -jar /app.jar" ]
```

Comment:
We could do it this way, but I would *much* prefer to have a separate script, to make it easier to add input sanitizing code, rather than doing `sh -c` and an inline script.

Reply:
I prefer doing it this way Jonathan, because of simplicity, that's for one, number two, if I try passing the endpoints in run time, I'll have to do so in the cloud formation template which is basically the same thing, it's going to be embedded in the CF template. the thing that is more complicated than the Dockerfile. So I'll try  to generate or customize a multiline shell script from a Jenkins job in Groovy and see what if it'll work, if not we'll stick with this one, for the sake of simplicity and time of course .

Reply:
To me, "simplicity" comes *firmly* down on the side of making it a separate script, whether parameters are passed in at runtime or edited into it at build time.

CloudFormation templates can be made to take parameters, and we could pass the information in via named environment variables instead of command-line parameters (see [this StackOverflow answer](https://stackoverflow.com/a/46504557) for what looks like a good starting point).

Reply:
I'll check it later, I'll try to  make the side script in the Jenkins Groovy and pass it in build time.

Reply:
The script can also be simply committed to version control like the `Dockerfile` itself.